node/hyperbahn/rate_limiting.js

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
```

Comment:
Like a relay_handler, a rate_limiting_handler is not necessarily coupled to Hyperbahn. Let’s move this into the tchannel/node directory. Let’s also break this into two parts, RateLimiter and RateLimitingHandler, both decoupled from Hyperbahn. It may not even be necessary to look up RateLimiter by service if there is one RateLimiter and RateLimitingHandler per service proxy.

Breaking RateLimiter out will make it easier to experiment, test, and reuse. RateLimiters could be used in other scenarios.